src/pages/LoginPage.js

```diff
@@ -91,8 +96,11 @@ class LoginPage extends React.Component {
                                                     value={password}
                                                     onChange={this.handleChange}/>
                                     </Form.Field>
-                                    <Button color='blue' fluid type='submit' size='large'>Login</Button>
-                                </Segment>
+                                    <Form.Group inline>
+                                        <Button color='blue' fluid type='submit' size='large'>Login</Button>
+                                        <Button color='blue' fluid type='button' onClick={this.handleRegister()} size='large'>Register</Button>
```

Comment:
Make this 'this.handleRegister' aka remove the ().  The () means it is called/evaluated at compile time which is causing a redirect everytime.